spire/humbug/api.py

```diff
@@ -407,33 +409,52 @@ async def delete_restricted_token_handler(
 
 
 @app.post("/reports", tags=["reports"], response_model=None)
-async def create_reports_handler(
-    request: Request,
-    report: HumbugReport,
-    db_session: Session = Depends(db.yield_connection_from_env),
-) -> Response:
+async def push_report_to_cache(request: Request, report: HumbugReport,) -> Response:
     """
-    Add report to integration journal.
-
-    - **title** (string): Entry title
-    - **content** (string): Entry content
-    - **tags** (list): Entry tags
+    Add report task to redis cache.
+
+    report task:
+        - report:
+            - **title** (string): Entry title
+            - **content** (string): Entry content
+            - **tags** (list): Entry tags
+        - **bugout_token** (UUID): Humbug token
     """
-    restricted_token = request.state.token
-    try:
-        journal_id = await actions.get_journal_id_by_restricted_token(
-            db_session, restricted_token=restricted_token
+    report.tags.append(f"reporter_token:{str(request.state.token)}")
+    report.tags = sorted(list(set(report.tags)))
+
+    with db.yield_redis_env_ctx() as redis_client:
+
+        redis_client.rpush(
+            "reports_queue",
+            HumbugCreateReportTask(
+                report=report, bugout_token=request.state.token
+            ).json(),
         )
 
-        await actions.create_report(
-            restricted_token=restricted_token, journal_id=journal_id, report=report
+    return Response(status_code=200)
+
+
+@app.post("/reports_bulk", tags=["reports"], response_model=None)
+async def bulk_push_reports_to_cache(
+    request: Request, reports_list: List[HumbugReport],
+) -> Response:
+
+    reports_pack = []
+
+    for report in reports_list:
+        report.tags.append(f"reporter_token:{str(request.state.token)}")
```

Comment:
We will move this to drones.

Our goal in Spire is to return a 200 response to user AS QUICKLY AS POSSIBLE.